tests/test_daemon.cpp

```diff
@@ -367,6 +368,23 @@ TEST_F(Daemon, failed_restart_command_returns_fulfilled_promise)
     EXPECT_TRUE(is_ready(status_promise.get_future()));
 }
 
+TEST_F(Daemon, proxy_contains_valid_info)
+{
+    QString username{"username"};
+    QString password{"password"};
+    QString hostname{"192.168.1.1"};
+    qint16 port{3128};
+    QString proxy = QString("%1:%2@%3:%4").arg(username).arg(password).arg(hostname).arg(port);
+
+    mpt::SetEnvScope env("http_proxy", proxy.toUtf8());
+
+    auto config = config_builder.build();
+
+    EXPECT_THAT(config->network_proxy->user(), username);
+    EXPECT_THAT(config->network_proxy->password(), password);
+    EXPECT_THAT(config->network_proxy->hostName(), hostname);
+    EXPECT_THAT(config->network_proxy->port(), port);
+}
```

Comment:
Thank you again. I have one last little concern nagging me :worried: This not only sets the application proxy for all ensuing tests, after the config gets out of scope, Qt could be left with a reference to an invalid object (the implementation probably copies, but we can't rely on that). 

I suggested extracting the call to `QNetworkProxy::setApplicationProxy` so that it did not run in the tests (on a quick look, I did not see a way to undo that).

Reply:
Hey @ricab,

Good catch!  I fixed this up, so it shouldn't be any concern now.

Reply:
Perfect!